pkg/reconciler/ingress/resources/gateway.go

```diff
@@ -72,13 +72,17 @@ func GetHTTPServer(gateway *v1alpha3.Gateway) *v1alpha3.Server {
 }
 
 func belongsToClusterIngress(server *v1alpha3.Server, ia v1alpha1.IngressAccessor) bool {
-	// The format of the portName should be "<clusteringress-name>:<number>".
-	// For example, route-test:0.
+	// The format of the portName should be "<namespace>-<ingress_name>:<number>".
+	// For example, default-routetest:0.
 	portNameSplits := strings.Split(server.Port.Name, ":")
 	if len(portNameSplits) != 2 {
 		return false
 	}
-	return portNameSplits[0] == ia.GetName()
+	// TODO: This should be removed after ClusterIngress codes are cleaned up.
+	if len(ia.GetNamespace()) == 0 {
+		return portNameSplits[0] == ia.GetName()
+	}
+	return portNameSplits[0] == ia.GetNamespace()+"-"+ia.GetName()
```

Comment:
is this change backwards compatible? What do with existing `Server` objects?

Reply:
Yes, I am sure that it is backwards compatible. 
The server objects and portname for Gateway are reconciled. And the legacy portnme servers are replaced with the portname with namespace, when it is reconciled without remaining.